src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1332,24 +1327,24 @@ void ProcessCreateInstanceByName (ref ReflectionPatternContext reflectionContext
 						if (typeNameValue is KnownStringValue typeNameStringValue) {
 							var resolvedAssembly = _context.GetLoadedAssembly (assemblyNameStringValue.Contents);
 							if (resolvedAssembly == null) {
-								reflectionContext.RecordUnrecognizedPattern ($"Activator call '{reflectionContext.MemberWithRequirements}' inside '{reflectionContext.Source}' references assembly '{assemblyNameStringValue.Contents}' which could not be found");
+								reflectionContext.RecordUnrecognizedPattern (2056, $"The assembly '{assemblyNameStringValue.Contents}' can not be found");
 								continue;
 							}
 
 							var resolvedType = resolvedAssembly.FindType (typeNameStringValue.Contents);
 							if (resolvedType == null) {
-								reflectionContext.RecordUnrecognizedPattern ($"Activator call '{reflectionContext.MemberWithRequirements}' inside '{reflectionContext.Source}' references type '{typeNameStringValue}' in assembly '{resolvedAssembly.FullName}' which could not be found");
+								reflectionContext.RecordUnrecognizedPattern (2057, $"The type '{typeNameStringValue.Contents}' can not be found in assembly '{resolvedAssembly.FullName}'");
```

Comment:
Why do we treat unresolved type as a warning for `CreateInstance` but not for `GetType`?

Reply:
Short - no idea - but thinking about it, it does make sense:
* GetType returns null when the type can't be found - so it's reasonable to consider non-failure cases (ignoring the overload which can throw... at least for now)
* CreateInstance throws if the type can't be found - so it makes more sense to treat this as potentially dangerous

But I do agree that this is a tricky thing to get right - I could see us going either way.

Reply:
> But I do agree that this is a tricky thing to get right - I could see us going either way.

We should only warn if there is risk trimming will change runtime behavior.

Trying to be helpful like this ("hey, I think you have a bug here") has resulted in support calls whenever we tried that. People have dead code (e.g. unreachable platform specific code) in their apps all the time. These warnings pretty much always find the dead code. If it was live code, people would have noticed. The code is "safe" for trimming purposes and that's all that should matter for linker.

I would vote to get rid of this warning.

I also filed #1387 because `FindType` is... inadequate.

Reply:
I'll remove this warning